packages/cli/src/commands/sandbox/sandbox-secret/sandbox_secret_set_command.ts

```diff
@@ -57,6 +60,28 @@ export class SandboxSecretSetCommand
       demandOption: true,
     });
   };
+
+  /**
+   * Prompt (or) read secret value from stdin based on terminal interactive mode
+   */
+  private readSecretValue = async (): Promise<string> => {
+    let secretValue = '';
+    if (this.readStream.isTTY) {
+      // This input is for interactive mode.
+      secretValue = await AmplifyPrompter.secretValue();
+    } else {
+      // This allows to accept secret value from redirected input `|` and `>`.
+      this.readStream.on('readable', () => {
+        const chunk = this.readStream.read();
+        if (chunk !== null) {
+          secretValue += chunk;
+        }
+      });
+      // Wait for the end of the input.
+      await once(this.readStream, 'end');
+    }
+    return secretValue;
+  };
```

Comment:
How do we handle situation where somebody called this comment in non-TTY session and didn't pipe anything to stdin ? 

Is this handled already by validation elsewhere or should we add something here?

Reply:
If someone has opened in non-TTY and hasn't written any data to the input stream and the input stream is in readable state, then the process would wait until it receives `end` event. It is intentional so that customers can pipe long running command's (such as an API call) output stream to set secret command.

In case they want to cancel the command, they can always pass SIGINT and terminate the process.

Reply:
I see. What might be worth considering is to add some timeout and throw error if data is not provided within reasonable interval.
Hanging forever is not great.

Reply:
I will take a look at how other bash commands handle this and come up with a follow-up.